src/layer/AbstractLayer.ts

```diff
@@ -31,42 +32,47 @@ export class AbstractLayer {
   }
 
   public async getMap(params: GetMapParams, api: ApiType, reqConfig?: RequestConfiguration): Promise<Blob> {
-    switch (api) {
-      case ApiType.WMS:
-        // When API type is set to WMS, getMap() uses getMapUrl() with the same provided parameters for
-        //   getting the url of the image.
-        // getMap() changes the received image according to provided gain and gamma after it is received.
-        // An error is thrown in getMapUrl() in case gain and gamma are present, because:
-        // - we don't send gain and gamma to the services as they may not be supported there and we don't want
-        //    to deceive the users with returning the image where gain and gamma were ignored
-        // - if they are supported on the services, gain and gamma would be applied twice in getMap() if they
-        //    were sent to the services in getMapUrl()
-        // In other words, gain and gamma need to be removed from the parameters in getMap() so the
-        //   errors in getMapUrl() are not triggered.
-        const paramsWithoutEffects = { ...params };
-        delete paramsWithoutEffects.gain;
-        delete paramsWithoutEffects.gamma;
-        const url = this.getMapUrl(paramsWithoutEffects, api);
-        const requestConfig: AxiosRequestConfig = {
-          // 'blob' responseType does not work with Node.js:
-          responseType: typeof window !== 'undefined' && window.Blob ? 'blob' : 'arraybuffer',
-          useCache: true,
-          ...getAxiosReqParams(reqConfig),
-        };
-        const response = await axios.get(url, requestConfig);
-        let blob = response.data;
+    const getMapValue = await ensureTimeout(async innerReqConfig => {
```

Comment:
Honestly, I have no idea how to name this, for me both `getMapValue` and `result` are bad - they do not really communicate what is being stored in the variable.

Since the switch always return a blob or throws an error, I renamed it to `blob` for now, but I am open to suggestions 😄